python/graph_coloring.py

```diff
@@ -0,0 +1,176 @@
+            Parameter.QA_TEMPERATURE: parameters.get(Parameter.QA_TEMPERATURE.value, 1),
+            Parameter.QA_MAX_STEPS: parameters.get(Parameter.QA_MAX_STEPS.value, 10),
+            Parameter.CONFLICT_ERR_ALPHA: parameters.get(
+                Parameter.CONFLICT_ERR_ALPHA.value, 1
+            ),
+            Parameter.CONFLICT_ERR_BETA: parameters.get(
+                Parameter.CONFLICT_ERR_BETA.value, 5
+            ),
+            Parameter.MUTATION: parameters.get(
+                Parameter.MUTATION.value, "SimpleMutation"
+            ),
+            Parameter.MULTIPLE_MUTATION_NODES_NO_OF_NODES: parameters.get(
+                Parameter.MULTIPLE_MUTATION_NODES_NO_OF_NODES.value, 2
+            ),
+            Parameter.RANDOM_MUTATION_PROBABILITY: parameters.get(
+                Parameter.RANDOM_MUTATION_PROBABILITY.value, 0.1
+            ),
+            Parameter.SIMPLE_TUNNELING_MUTATION: parameters.get(
+                Parameter.SIMPLE_TUNNELING_MUTATION.value, "MultipleMutation"
+            ),
+            Parameter.SIMPLE_TUNNELING_PROBABILITY: parameters.get(
+                Parameter.SIMPLE_TUNNELING_PROBABILITY.value, 0.5
+            ),
+            Parameter.SIMPLE_TUNNELING_ERROR_CORRECTION: parameters.get(
+                Parameter.SIMPLE_TUNNELING_ERROR_CORRECTION.value, 2
+            ),
+            Parameter.SIMPLE_TUNNELING_MAX_ATTEMPTS: parameters.get(
+                Parameter.SIMPLE_TUNNELING_MAX_ATTEMPTS.value, 25
+            ),
+            Parameter.CONVERGENCE_CALLBACK_TOLERANCE: parameters.get(
+                Parameter.CONVERGENCE_CALLBACK_TOLERANCE.value, 100
+            ),
+            Parameter.CONVERGENCE_CALLBACK_ACTIONS: parameters.get(
+                Parameter.CONVERGENCE_CALLBACK_ACTIONS.value, ["SimpleTunneling"]
+            ),
+        }
+    )
+    alg = params[Parameter.ALGORITHM]
+    sol = alg.run(graph, params)
+    return sol.chromosome
+
+
+def _convert_to_graph(context: mgp.ProcCtx) -> Graph:
+    nodes = []
+    adj_list = defaultdict(list)
+
+    for v in context.graph.vertices:
+        context.check_must_abort()
+        nodes.append(v.id)
+
+    for v in context.graph.vertices:
+        context.check_must_abort()
+        for e in v.out_edges:
+            weight = e.properties.get("weight", 1)
+            adj_list[e.from_vertex.id].append((e.to_vertex.id, weight))
+            adj_list[e.to_vertex.id].append((e.from_vertex.id, weight))
+
+    return Graph(nodes, adj_list)
+
+
+def _convert_to_subgraph(
+    context: mgp.ProcCtx, vertices: mgp.List[mgp.Vertex], edges: mgp.List[mgp.Edge]
+) -> Optional[Graph]:
+
+    vertices, edges = map(set, [vertices, edges])
+
+    nodes = []
+    adj_list = defaultdict(list)
+
+    for v in vertices:
+        context.check_must_abort()
+        nodes.append(v.id)
+
+    for e in edges:
+        context.check_must_abort()
+        weight = e.properties.get("weight", 1)
```

Comment:
This might as well be another module parameter (the name of property between nodes). Otherwise this is great solution by defaulting to 1.